lib/resty/iprepd.lua

```diff
@@ -62,6 +63,10 @@ function _M.check(self, ip)
         ngx.log(ngx.ERR, 'Unable to parse `reputation` value from response body')
       end
     else
+      if self.cache_errors == 1 then
+        self.cache:set(ip, 100, self.cache_ttl)
```

Comment:
should we emit a log message when we are caching failures just to make it more obvious this is happening?

Reply:
Actually, re-reading your description -- so with this we will cache 404's correct? This comment is probably not valid then, since 404 isn't really an error condition (it just indicates that iprepd doesn't know anything about the IP address)

Unless we want to specifically handle 404 (which is likely going to be seen the majority of the time in requests) differently than other 4xx or 5xx errors